src/hb/ot_layout_gsubgpos.rs

```diff
@@ -977,7 +980,7 @@ fn apply_lookup(
         }
     }
 
-    ctx.buffer.move_to(end);
+    ctx.buffer.move_to(end as usize);
```

Comment:
`end` is always guaranteed to be positive here?

Reply:
Good question. It looks like on this point it is a bug-for-bug port of HB. The upstream function takes an unsigned integer argument:

```c
bool
hb_buffer_t::move_to (unsigned int i)
```

But it is being *passed* a signed integer:

```c
int end;
...
(void) buffer->move_to (end);
```

I don't know if/how C resolves this. @behdad?

At some point it looks like there is a guarantee that it will be at least as large as...

```c
if (end < int (match_positions[idx]))
{
  end = match_positions[idx];
```

Reply:
Yes, `end` is also always positive because there's at least one glyph in the match sequence... You can assert as you cast it.

Would you be able to send a PR to HB to add similar asserts?

Reply:
I added an assertion for this case too (via Rust's TryInto trait) and also opened a [matching PR on HB](https://github.com/harfbuzz/harfbuzz/pull/4934).